pkgs/tools/networking/mailctl/default.nix

```diff
@@ -0,0 +1,85 @@
+{ mkDerivation
+, fetchFromSourcehut
+, aeson
+, base
+, bytestring
+, containers
+, directory
+, hsyslog
+, http-conduit
+, lib
+, network-uri
+, optparse-applicative
+, pretty-simple
+, process
+, template-haskell
+, text
+, time
+, twain
+, utf8-string
+, warp
+, yaml
+}:
+mkDerivation rec {
+  pname = "mailctl";
+  version = "0.8.8";
+
+  src = fetchFromSourcehut {
+    owner = "~petrus";
+    repo = "mailctl";
+    rev = version;
+    hash = "sha256-aFt6y2DzreROLcOLU8ynnSSVQW840T5wFqSRdSODQX4=";
+  };
+
+  isLibrary = true;
+  isExecutable = true;
+
+  libraryHaskellDepends = [
+    aeson
+    base
+    bytestring
+    containers
+    directory
+    hsyslog
+    http-conduit
+    network-uri
+    optparse-applicative
+    pretty-simple
+    process
+    template-haskell
+    text
+    time
+    twain
+    utf8-string
+    warp
+    yaml
+  ];
+
+  executableHaskellDepends = [
+    aeson
+    base
+    bytestring
+    containers
+    directory
+    hsyslog
+    http-conduit
+    network-uri
+    optparse-applicative
+    pretty-simple
+    process
+    template-haskell
+    text
+    time
+    twain
+    utf8-string
+    warp
+    yaml
+  ];
+
+  description = "OAuth2 tool for mail clients";
+  homepage = "https://sr.ht/~petrus/mailctl/";
+  changelog = "https://git.sr.ht/~petrus/mailctl/refs/0.8.8";
```

Comment:
```suggestion
  changelog = "https://git.sr.ht/~petrus/mailctl/refs/${version}";
```

Reply:
I've tried doing something like this before, and it broke ofborg eval, I think because it was IFD.  I'll try and see how it goes.

Reply:
Why do you think it's an IFD ?

Reply:
I'm probably misremembering, or misunderstood at the time.